src/libsync/syncfilestatustracker.cpp

```diff
@@ -157,6 +162,20 @@ SyncFileStatus SyncFileStatusTracker::fileStatus(const QString& systemFileName)
     return SyncFileStatus();
 }
 
+void SyncFileStatusTracker::slotPathTouched(const QString& systemFileName)
+{
+    QString fileName = systemFileName.normalized(QString::NormalizationForm_C);
+
+    QString folderPath = _syncEngine->localPath();
+    if (!fileName.startsWith(folderPath))
```

Comment:
No we don't ship with assert enabled, in that case it will either break something or not work properly, only developers would get the sig abort if something is wrong.

That's however a risk that we should usually be comfortable taking, it forces us to patch a whole product and not just a single file. No strong opinion however, so feel free to follow your guts.